velox/docs/functions/spark/array.rst

```diff
@@ -126,6 +126,14 @@ Array Functions
     if :doc:`spark.legacy_size_of_null <../../configs>` is set to false.
     Otherwise, returns -1 for null input.
 
+.. function:: slice(array(E), start, length) -> array(E)
+
+    Returns a subarray starting from index ``start``(or starting from the end
+    if ``start`` is negative) with a length of ``length``.
+
+        SELECT slice(array(1, 2, 3, 4), 2, 2); -- [2,3]
```

Comment:
Does the index start from 1? Maybe document that if that's the case. And what is the behavior when start is 0?

Reply:
Yeap, start from 1, it will throw exception if start is 0:

```c++
if (UNLIKELY(index == 0)) {
      VELOX_USER_FAIL("SQL array indices start at 1");
    }
```

Reply:
Would you add a test to verify this exception?